api/services/SQS.js

```diff
@@ -1,82 +1,81 @@
-            { "name": "AWS_ACCESS_KEY_ID", "value": awsKey },
-            { "name": "AWS_SECRET_ACCESS_KEY", "value": awsSecret },
-            { "name": "CALLBACK", "value": tokens.callback },
-            { "name": "BUCKET", "value":  sails.config.build.s3Bucket },
-            { "name": "BASEURL", "value": tokens.baseurl },
-            { "name": "CACHE_CONTROL", "value": sails.config.build.cacheControl },
-            { "name": "BRANCH", "value": tokens.branch },
-            { "name": "CONFIG", "value": tokens.config },
-            { "name": "REPOSITORY", "value": tokens.repository },
-            { "name": "OWNER", "value": tokens.owner },
-            { "name": "PREFIX", "value": tokens.prefix },
-            { "name": "GITHUB_TOKEN", "value": tokens.token },
-            { "name": "GENERATOR", "value": tokens.engine }
-          ],
-          name: sails.config.build.containerName
-        },
-        params = { QueueUrl: queueUrl };
+var buildContainerEnvironment = (build) => ({
+  AWS_DEFAULT_REGION: buildConfig.awsRegion,
+  AWS_ACCESS_KEY_ID: buildConfig.awsBuildKey,
+  AWS_SECRET_ACCESS_KEY: buildConfig.awsBuildSecret,
+  CALLBACK: `${buildConfig.callback}${build.id}/${buildConfig.token}`,
+  BUCKET: buildConfig.s3Bucket,
+  BASEURL: baseURLForBuild(build),
+  CACHE_CONTROL: buildConfig.cacheControl,
+  BRANCH: build.branch,
+  CONFIG: build.site.config,
+  REPOSITORY: build.site.repository,
+  OWNER: build.site.owner,
+  PREFIX: pathForBuild(build),
+  GITHUB_TOKEN: githubTokenForBuild(build),
+  GENERATOR: build.site.engine,
+  SOURCE_REPO: sourceForBuild(build).repository,
+  SOURCE_OWNER: sourceForBuild(build).owner
+})
 
-    if (tokens.sourceRepo) body.environment.push({
-      "name": "SOURCE_REPO",
-      "value": tokens.sourceRepo
-    });
-    if (tokens.sourceOwner) body.environment.push({
-      "name": "SOURCE_OWNER",
-      "value": tokens.sourceOwner
-    });
-
-    params.MessageBody = JSON.stringify(body);
+var defaultBranch = (build) => {
+  return build.branch === build.site.defaultBranch
+}
 
-    sails.log.verbose('SQS payload build', tokens.callback, sails.config.build.s3Bucket, awsKey);
+var pathForBuild = (build) => {
+  if (defaultBranch(build)) {
+    return `site/${build.site.owner}/${build.site.repository}`
+  } else {
+    return `preview/${build.site.owner}/${build.site.repository}/${build.branch}`
+  }
+}
 
-    sqs.sendMessage(params, function(err, data) {
-      sails.log.verbose('message sent with error? and data', err, data);
-      if (err) error(err, model);
-    });
+var baseURLForBuild = (build) => {
+  if (defaultBranch(build) && build.site.domain) {
+    return ""
+  } else {
+    return "/" + pathForBuild(build)
+  }
+}
 
+var githubTokenForBuild = (build) => {
+  if (build.user.passport) {
+    return build.user.passport.tokens.accessToken
+  } else {
+    return ""
   }
+}
 
-};
+var sourceForBuild = (build) => {
+  return build.source || {}
+}
+
+SQS.buildMessageBody = (build) => {
```

Comment:
well, if we move this (since it's really about what the build container wants to see and not how we use SQS), then everyone should be happier.